openmls/src/key_packages/mod.rs

```diff
@@ -401,6 +401,11 @@ impl KeyPackage {
     pub fn last_resort(&self) -> bool {
         self.payload.extensions.contains(ExtensionType::LastResort)
     }
+
+    /// Get the lifetime of the KeyPackage
+    pub fn life_time(&self) -> &Lifetime {
+        self.payload.leaf_node.life_time().unwrap()
```

Comment:
I think the `Option` comes from the fact that lifetime is an extension in the list of extensions of a leaf node. The spec mandates its presence, but it's not enforced by the type system from the spec. I wonder if it wouldn't be a good idea to refactor LeafNode so that there is no `Option` for mandated extensions. Having an `unwrap()` at this place feels somewhat brittle.

Reply:
The only idea I have here is to have per-source types of `LeafNode` (e.g. `KeyPackageLeafNode`, `UpdateLeafNode`) that are used in the respective places, make `LeafNode` an enum over these types and implement `From<KeyPackageLeafNode>` (and others) for `LeafNode`.

Reply:
I added an issue for this: https://github.com/openmls/openmls/issues/1663